Building python2 for armeabi fails due to space character in storage_dir (OS X)

`storage_dir` will be `/Users/username/Library/Application Support/python-for-android` on OS X

```
[INFO]:    This python-for-android revamp is an experimental alpha release!
[INFO]:    It should work (mostly), but you may experience missing features or bugs.
[INFO]:    Will compile for the following archs: armeabi
[INFO]:    Found possible SDK dirs in buildozer dir: android-sdk-20, android-sdk-21
[INFO]:    Will attempt to use SDK at /Users/ben/.buildozer/android/platform/android-sdk-20
[WARNING]: This SDK lookup is intended for debug only, if you use python-for-android much you should probably maintain your own SDK download.
[INFO]:    Getting Android API version from user argument
[INFO]:    Available Android APIs are (19)
[INFO]:    Requested API target 19 is available, continuing.
[INFO]:    Found possible NDK dirs in buildozer dir: android-ndk-r9c, android-ndk-r9c-darwin-x86_64.tar.bz2
[INFO]:    Will attempt to use NDK at /Users/ben/.buildozer/android/platform/android-ndk-r9c
[WARNING]: This NDK lookup is intended for debug only, if you use python-for-android much you should probably maintain your own NDK download.
[INFO]:    Got NDK version from $ANDROIDNDKVER
[INFO]:    Got Android NDK version from the NDK dir: it is r9c
[INFO]:    Using Google NDK r9c
[INFO]:    Found virtualenv at /usr/local/bin/virtualenv-2.7
[INFO]:    ccache is missing, the build will not be optimized in the future.
[INFO]:    Found the following toolchain versions: ['4.6', '4.8', 'clang3.3']
[INFO]:    Picking the latest gcc toolchain, here 4.8
[WARNING]: Missing executable: libtoolize is not installed
[INFO]:    No existing dists meet the given requirements!
[INFO]:    No dist exists that meets your requirements, so one will be built.
[INFO]:    Bootstrap requires recipes ['sdl2', ('python2', 'python3crystax')]
[INFO]:    Loaded recipe sdl2 (depends on [('python2', 'python3crystax'), 'sdl2_image', 'sdl2_mixer', 'sdl2_ttf'], conflicts ['sdl', 'pygame', 'pygame_bootstrap_components'])
[INFO]:    Loaded recipe hostpython2 (depends on [], conflicts ['hostpython3'])
[INFO]:    Loaded recipe kivy (depends on ['pyjnius', ('sdl2', 'pygame'), ('python2', 'python3crystax')])
[INFO]:    Loaded recipe sdl2_image (depends on [])
[INFO]:    Loaded recipe sdl2_mixer (depends on [])
[INFO]:    Loaded recipe sdl2_ttf (depends on [])
[INFO]:    Loaded recipe pyjnius (depends on ['six', ('sdl2', 'sdl'), ('python2', 'python3crystax')])
[INFO]:    Loaded recipe six (depends on [('python2', 'python3crystax')])
[INFO]:    Found multiple valid recipe sets:
[INFO]:        ['six', 'sdl2', u'kivy', 'sdl2_ttf', u'hostpython2', 'sdl2_mixer', 'sdl2_image', 'pyjnius', 'python2']
[INFO]:        ['sdl2', 'six', 'sdl2_ttf', u'hostpython2', 'python3crystax', 'sdl2_mixer', 'sdl2_image', 'pyjnius', u'kivy']
[INFO]:    Using the first of these: ['six', 'sdl2', u'kivy', 'sdl2_ttf', u'hostpython2', 'sdl2_mixer', 'sdl2_image', 'pyjnius', 'python2']
[INFO]:    The selected bootstrap is sdl2
[INFO]:    # Creating dist with sdl2 bootstrap
[INFO]:    Dist will have name unnamed_dist_7 and recipes (kivy, hostpython2)
[INFO]:    -> running cp -r /usr/local/lib/python2.7/site-packages/pythonforandroid/bootst...(and 109 more)
[INFO]:    -> directory context /Users/ben/Library/Application Support/python-for-android/build/bootstrap_builds/sdl2-python2
[INFO]:    <- directory context /private/tmp/test
[INFO]:    Recipe build order is [u'hostpython2', 'python2', 'sdl2_image', 'sdl2_mixer', 'sdl2_ttf', 'sdl2', 'six', 'pyjnius', u'kivy']
[INFO]:    # Downloading recipes
[INFO]:    Downloading hostpython2
[INFO]:    -> running mkdir -p /Users/ben/Library/Application Support/python-for-android/p...(and 19 more)
[INFO]:    -> directory context /Users/ben/Library/Application Support/python-for-android/packages/hostpython2
[INFO]:    -> running basename http://python.org/ftp/python/2.7.2/Python-2.7.2.tar.bz2
[INFO]:    hostpython2 download already cached, skipping
[WARNING]: Should check headers here! Skipping for now.
[INFO]:    <- directory context /private/tmp/test
[INFO]:    Downloading python2
[INFO]:    -> running mkdir -p /Users/ben/Library/Application Support/python-for-android/packages/python2
[INFO]:    -> directory context /Users/ben/Library/Application Support/python-for-android/packages/python2
[INFO]:    -> running basename http://python.org/ftp/python/2.7.2/Python-2.7.2.tar.bz2
[INFO]:    python2 download already cached, skipping
[WARNING]: Should check headers here! Skipping for now.
[INFO]:    <- directory context /private/tmp/test
[INFO]:    Downloading sdl2_image
[INFO]:    -> running mkdir -p /Users/ben/Library/Application Support/python-for-android/packages/sdl2_image
[INFO]:    -> directory context /Users/ben/Library/Application Support/python-for-android/packages/sdl2_image
[INFO]:    -> running basename https://www.libsdl.org/projects/SDL_image/release/SDL2_image-2.0.0.tar.gz
[INFO]:    sdl2_image download already cached, skipping
[WARNING]: Should check headers here! Skipping for now.
[INFO]:    <- directory context /private/tmp/test
[INFO]:    Downloading sdl2_mixer
[INFO]:    -> running mkdir -p /Users/ben/Library/Application Support/python-for-android/packages/sdl2_mixer
[INFO]:    -> directory context /Users/ben/Library/Application Support/python-for-android/packages/sdl2_mixer
[INFO]:    -> running basename https://www.libsdl.org/projects/SDL_mixer/release/SDL2_mixer-2.0.0.tar.gz
[INFO]:    sdl2_mixer download already cached, skipping
[WARNING]: Should check headers here! Skipping for now.
[INFO]:    <- directory context /private/tmp/test
[INFO]:    Downloading sdl2_ttf
[INFO]:    -> running mkdir -p /Users/ben/Library/Application Support/python-for-android/packages/sdl2_ttf
[INFO]:    -> directory context /Users/ben/Library/Application Support/python-for-android/packages/sdl2_ttf
[INFO]:    -> running basename https://www.libsdl.org/projects/SDL_ttf/release/SDL2_ttf-2.0.12.tar.gz
[INFO]:    sdl2_ttf download already cached, skipping
[WARNING]: Should check headers here! Skipping for now.
[INFO]:    <- directory context /private/tmp/test
[INFO]:    Downloading sdl2
[INFO]:    -> running mkdir -p /Users/ben/Library/Application Support/python-for-android/packages/sdl2
[INFO]:    -> directory context /Users/ben/Library/Application Support/python-for-android/packages/sdl2
[INFO]:    -> running basename https://www.libsdl.org/release/SDL2-2.0.3.tar.gz
[INFO]:    sdl2 download already cached, skipping
[WARNING]: Should check headers here! Skipping for now.
[INFO]:    <- directory context /private/tmp/test
[INFO]:    Downloading six
[INFO]:    -> running mkdir -p /Users/ben/Library/Application Support/python-for-android/packages/six
[INFO]:    -> directory context /Users/ben/Library/Application Support/python-for-android/packages/six
[INFO]:    -> running basename https://pypi.python.org/packages/source/s/six/six-1.9.0.tar.gz
[INFO]:    six download already cached, skipping
[WARNING]: Should check headers here! Skipping for now.
[INFO]:    <- directory context /private/tmp/test
[INFO]:    Downloading pyjnius
[INFO]:    -> running mkdir -p /Users/ben/Library/Application Support/python-for-android/packages/pyjnius
[INFO]:    -> directory context /Users/ben/Library/Application Support/python-for-android/packages/pyjnius
[INFO]:    -> running basename https://github.com/kivy/pyjnius/archive/master.zip
[INFO]:    pyjnius download already cached, skipping
[WARNING]: Should check headers here! Skipping for now.
[INFO]:    <- directory context /private/tmp/test
[INFO]:    Downloading kivy
[INFO]:    -> running mkdir -p /Users/ben/Library/Application Support/python-for-android/packages/kivy
[INFO]:    -> directory context /Users/ben/Library/Application Support/python-for-android/packages/kivy
[INFO]:    -> running basename https://github.com/kivy/kivy/archive/master.zip
[INFO]:    kivy download already cached, skipping
[WARNING]: Should check headers here! Skipping for now.
[INFO]:    <- directory context /private/tmp/test
[INFO]:    # Building all recipes for arch armeabi
[INFO]:    # Unpacking recipes
[INFO]:    Unpacking hostpython2 for armeabi
[INFO]:    -> running basename http://python.org/ftp/python/2.7.2/Python-2.7.2.tar.bz2
[INFO]:    -> directory context /Users/ben/Library/Application Support/python-for-android/build/other_builds/hostpython2/desktop
[INFO]:    hostpython2 is already unpacked, skipping
[INFO]:    <- directory context /private/tmp/test
[INFO]:    Unpacking python2 for armeabi
[INFO]:    -> running basename http://python.org/ftp/python/2.7.2/Python-2.7.2.tar.bz2
[INFO]:    -> directory context /Users/ben/Library/Application Support/python-for-android/build/other_builds/python2/armeabi
[INFO]:    python2 is already unpacked, skipping
[INFO]:    <- directory context /private/tmp/test
[INFO]:    Unpacking sdl2_image for armeabi
[INFO]:    -> running basename https://www.libsdl.org/projects/SDL_image/release/SDL2_image-2.0.0.tar.gz
[INFO]:    -> directory context /Users/ben/Library/Application Support/python-for-android/build/bootstrap_builds/sdl2-python2/jni
[INFO]:    sdl2_image is already unpacked, skipping
[INFO]:    <- directory context /private/tmp/test
[INFO]:    Unpacking sdl2_mixer for armeabi
[INFO]:    -> running basename https://www.libsdl.org/projects/SDL_mixer/release/SDL2_mixer-2.0.0.tar.gz
[INFO]:    -> directory context /Users/ben/Library/Application Support/python-for-android/build/bootstrap_builds/sdl2-python2/jni
[INFO]:    sdl2_mixer is already unpacked, skipping
[INFO]:    <- directory context /private/tmp/test
[INFO]:    Unpacking sdl2_ttf for armeabi
[INFO]:    -> running basename https://www.libsdl.org/projects/SDL_ttf/release/SDL2_ttf-2.0.12.tar.gz
[INFO]:    -> directory context /Users/ben/Library/Application Support/python-for-android/build/bootstrap_builds/sdl2-python2/jni
[INFO]:    sdl2_ttf is already unpacked, skipping
[INFO]:    <- directory context /private/tmp/test
[INFO]:    Unpacking sdl2 for armeabi
[INFO]:    -> running basename https://www.libsdl.org/release/SDL2-2.0.3.tar.gz
[INFO]:    -> directory context /Users/ben/Library/Application Support/python-for-android/build/bootstrap_builds/sdl2-python2/jni
[INFO]:    sdl2 is already unpacked, skipping
[INFO]:    <- directory context /private/tmp/test
[INFO]:    Unpacking six for armeabi
[INFO]:    -> running basename https://pypi.python.org/packages/source/s/six/six-1.9.0.tar.gz
[INFO]:    -> directory context /Users/ben/Library/Application Support/python-for-android/build/other_builds/six-python2/armeabi
[INFO]:    six is already unpacked, skipping
[INFO]:    <- directory context /private/tmp/test
[INFO]:    Unpacking pyjnius for armeabi
[INFO]:    -> running basename https://github.com/kivy/pyjnius/archive/master.zip
[INFO]:    -> directory context /Users/ben/Library/Application Support/python-for-android/build/other_builds/pyjnius-python2-sdl2/armeabi
[INFO]:    pyjnius is already unpacked, skipping
[INFO]:    <- directory context /private/tmp/test
[INFO]:    Unpacking kivy for armeabi
[INFO]:    -> running basename https://github.com/kivy/kivy/archive/master.zip
[INFO]:    -> directory context /Users/ben/Library/Application Support/python-for-android/build/other_builds/kivy-python2-sdl2/armeabi
[INFO]:    kivy is already unpacked, skipping
[INFO]:    <- directory context /private/tmp/test
[INFO]:    # Prebuilding recipes
[INFO]:    Prebuilding hostpython2 for armeabi
[INFO]:    -> running cp /usr/local/lib/python2.7/site-packages/pythonforandroid/recipes/h...(and 139 more)
[INFO]:    Prebuilding python2 for armeabi
[INFO]:    python2 has no prebuild_armeabi, skipping
[INFO]:    Applying patches for python2[armeabi]
[INFO]:    python2 already patched, skipping
[INFO]:    Prebuilding sdl2_image for armeabi
[INFO]:    sdl2_image has no prebuild_armeabi, skipping
[INFO]:    Applying patches for sdl2_image[armeabi]
[INFO]:    sdl2_image already patched, skipping
[INFO]:    Prebuilding sdl2_mixer for armeabi
[INFO]:    sdl2_mixer has no prebuild_armeabi, skipping
[INFO]:    Applying patches for sdl2_mixer[armeabi]
[INFO]:    sdl2_mixer already patched, skipping
[INFO]:    Prebuilding sdl2_ttf for armeabi
[INFO]:    sdl2_ttf has no prebuild_armeabi, skipping
[INFO]:    Prebuilding sdl2 for armeabi
[INFO]:    sdl2 has no prebuild_armeabi, skipping
[INFO]:    Applying patches for sdl2[armeabi]
[INFO]:    sdl2 already patched, skipping
[INFO]:    Prebuilding six for armeabi
[INFO]:    six has no prebuild_armeabi, skipping
[INFO]:    Prebuilding pyjnius for armeabi
[INFO]:    pyjnius has no prebuild_armeabi, skipping
[INFO]:    Applying patches for pyjnius[armeabi]
[INFO]:    pyjnius already patched, skipping
[INFO]:    Prebuilding kivy for armeabi
[INFO]:    kivy has no prebuild_armeabi, skipping
[INFO]:    # Building recipes
[INFO]:    Building hostpython2 for armeabi
[INFO]:    -> directory context /Users/ben/Library/Application Support/python-for-android/build/other_builds/hostpython2/desktop/hostpython2
[INFO]:    hostpython already exists, skipping build
[INFO]:    <- directory context /private/tmp/test
[INFO]:    Building python2 for armeabi
[INFO]:    -> running cp /Users/ben/Library/Application Support/python-for-android/build/o...(and 155 more)
[INFO]:    -> running cp /Users/ben/Library/Application Support/python-for-android/build/o...(and 153 more)
[INFO]:    -> directory context /Users/ben/Library/Application Support/python-for-android/build/other_builds/python2/armeabi/python2
[INFO]:    -> running cp /usr/local/lib/python2.7/site-packages/pythonforandroid/recipes/h...(and 24 more)
('path is', '/Users/ben/.buildozer/android/platform/android-ndk-r9c/toolchains/arm-linux-androideabi-4.8/prebuilt/darwin-x86/bin/:/Users/ben/.buildozer/android/platform/android-ndk-r9c/toolchains/arm-linux-androideabi-4.8/prebuilt/darwin-x86_64/bin/:/Users/ben/.buildozer/android/platform/android-ndk-r9c:/Users/ben/.buildozer/android/platform/android-sdk-20/tools:/usr/local/bin:/usr/local/sbin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin')
[INFO]:    -> running gcc -dumpmachine
[INFO]:    -> running configure --host=arm-eabi --build=x86_64-apple-darwin15.3.0 --prefix...(and 175 more)
Traceback (most recent call last):
  File "/usr/local/bin/p4a", line 9, in <module>
    load_entry_point('python-for-android==0.3', 'console_scripts', 'p4a')()
  File "/usr/local/lib/python2.7/site-packages/pythonforandroid/toolchain.py", line 708, in main
    ToolchainCL()
  File "/usr/local/lib/python2.7/site-packages/pythonforandroid/toolchain.py", line 323, in __init__
    getattr(self, args.command)(unknown)
  File "/usr/local/lib/python2.7/site-packages/pythonforandroid/toolchain.py", line 105, in wrapper_func
    build_dist_from_args(ctx, dist, dist_args)
  File "/usr/local/lib/python2.7/site-packages/pythonforandroid/toolchain.py", line 142, in build_dist_from_args
    build_recipes(build_order, python_modules, ctx)
  File "/usr/local/lib/python2.7/site-packages/pythonforandroid/build.py", line 552, in build_recipes
    recipe.build_arch(arch)
  File "/usr/local/lib/python2.7/site-packages/pythonforandroid/recipes/python2/__init__.py", line 44, in build_arch
    self.do_python_build(arch)
  File "/usr/local/lib/python2.7/site-packages/pythonforandroid/recipes/python2/__init__.py", line 114, in do_python_build
    _env=env)
  File "/usr/local/lib/python2.7/site-packages/pythonforandroid/logger.py", line 160, in shprint
    for line in output:
  File "/usr/local/lib/python2.7/site-packages/sh.py", line 565, in next
    self.wait()
  File "/usr/local/lib/python2.7/site-packages/sh.py", line 500, in wait
    self.handle_command_exit_code(exit_code)
  File "/usr/local/lib/python2.7/site-packages/sh.py", line 516, in handle_command_exit_code
    raise exc(self.ran, self.process.stdout, self.process.stderr)
sh.ErrorReturnCode_77:

  RAN: './configure --host=arm-eabi --build=x86_64-apple-darwin15.3.0 --prefix=/Users/ben/Library/Application Support/python-for-android/build/other_builds/python2/armeabi/python2/python-install --enable-shared --disable-toolbox-glue --disable-framework'

  STDOUT:
checking for --enable-universalsdk... no
checking for --with-universal-archs... 32-bit
checking MACHDEP... darwin
checking EXTRAPLATDIR... $(PLATMACDIRS)
checking machine type as reported by uname -m... x86_64
checking for --without-gcc... no
checking for arm-eabi-gcc... arm-linux-androideabi-gcc -DANDROID -mandroid -fomit-frame-pointer --sysroot /Users/ben/.buildozer/android/platform/android-ndk-r9c/platforms/android-19/arch-arm
checking whether the C compiler works... no
configure: error: in `/Users/ben/Library/Application Support/python-for-android/build/other_builds/python2/armeabi/python2':
configure: error: C compiler cannot create executables
See `config.log' for more details
```

config.log:
````
This file contains any messages produced by compilers while
running configure, to aid debugging if configure makes a mistake.

It was created by python configure 2.7, which was
generated by GNU Autoconf 2.68.  Invocation command line was

  $ ./configure --host=arm-eabi --build=x86_64-apple-darwin15.3.0 --prefix=/Users/ben/Library/Application Support/python-for-android/build/other_builds/python2/armeabi/python2/python-install --enable-shared --disable-toolbox-glue --disable-framework

## --------- ##
## Platform. ##
## --------- ##

hostname = Bens-Macbook.local
uname -m = x86_64
uname -r = 15.3.0
uname -s = Darwin
uname -v = Darwin Kernel Version 15.3.0: Thu Dec 10 18:40:58 PST 2015; root:xnu-3248.30.4~1/RELEASE_X86_64

/usr/bin/uname -p = i386
/bin/uname -X     = unknown

/bin/arch              = unknown
/usr/bin/arch -k       = unknown
/usr/convex/getsysinfo = unknown
/usr/bin/hostinfo      = Mach kernel version:
	 Darwin Kernel Version 15.3.0: Thu Dec 10 18:40:58 PST 2015; root:xnu-3248.30.4~1/RELEASE_X86_64
Kernel configured for up to 8 processors.
4 processors are physically available.
8 processors are logically available.
Processor type: i486 (Intel 80486)
Processors active: 0 1 2 3 4 5 6 7
Primary memory available: 8.00 gigabytes
Default processor set: 364 tasks, 1844 threads, 8 processors
Load average: 3.25, Mach factor: 4.95
/bin/machine           = unknown
/usr/bin/oslevel       = unknown
/bin/universe          = unknown

PATH: /Users/ben/.buildozer/android/platform/android-ndk-r9c/toolchains/arm-linux-androideabi-4.8/prebuilt/darwin-x86/bin/
PATH: /Users/ben/.buildozer/android/platform/android-ndk-r9c/toolchains/arm-linux-androideabi-4.8/prebuilt/darwin-x86_64/bin/
PATH: /Users/ben/.buildozer/android/platform/android-ndk-r9c
PATH: /Users/ben/.buildozer/android/platform/android-sdk-20/tools
PATH: /usr/local/bin
PATH: /usr/local/sbin
PATH: /usr/bin
PATH: /bin
PATH: /usr/sbin
PATH: /sbin
PATH: /opt/X11/bin


## ----------- ##
## Core tests. ##
## ----------- ##

configure:2764: checking for --enable-universalsdk
configure:2805: result: no
configure:2814: checking for --with-universal-archs
configure:2831: result: 32-bit
configure:2988: checking MACHDEP
configure:3136: result: darwin
configure:3142: checking EXTRAPLATDIR
configure:3157: result: $(PLATMACDIRS)
configure:3168: checking machine type as reported by uname -m
configure:3171: result: x86_64
configure:3184: checking for --without-gcc
configure:3228: result: no
configure:3249: checking for arm-eabi-gcc
configure:3276: result: arm-linux-androideabi-gcc -DANDROID -mandroid -fomit-frame-pointer --sysroot /Users/ben/.buildozer/android/platform/android-ndk-r9c/platforms/android-19/arch-arm
configure:3545: checking for C compiler version
configure:3554: arm-linux-androideabi-gcc -DANDROID -mandroid -fomit-frame-pointer --sysroot /Users/ben/.buildozer/android/platform/android-ndk-r9c/platforms/android-19/arch-arm --version >&5
arm-linux-androideabi-gcc (GCC) 4.8
Copyright (C) 2013 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

configure:3565: $? = 0
configure:3554: arm-linux-androideabi-gcc -DANDROID -mandroid -fomit-frame-pointer --sysroot /Users/ben/.buildozer/android/platform/android-ndk-r9c/platforms/android-19/arch-arm -v >&5
Using built-in specs.
COLLECT_GCC=arm-linux-androideabi-gcc
COLLECT_LTO_WRAPPER=/Users/ben/.buildozer/android/platform/android-ndk-r9c/toolchains/arm-linux-androideabi-4.8/prebuilt/darwin-x86_64/bin/../libexec/gcc/arm-linux-androideabi/4.8/lto-wrapper
Target: arm-linux-androideabi
Configured with: /s/ndk-toolchain/src/build/../gcc/gcc-4.8/configure --prefix=/tmp/ndk-User/build/toolchain/prefix --target=arm-linux-androideabi --host=x86_64-apple-darwin --build=x86_64-apple-darwin --with-gnu-as --with-gnu-ld --enable-languages=c,c++ --with-gmp=/tmp/ndk-User/build/toolchain/temp-install --with-mpfr=/tmp/ndk-User/build/toolchain/temp-install --with-mpc=/tmp/ndk-User/build/toolchain/temp-install --with-cloog=/tmp/ndk-User/build/toolchain/temp-install --with-isl=/tmp/ndk-User/build/toolchain/temp-install --with-ppl=/tmp/ndk-User/build/toolchain/temp-install --disable-ppl-version-check --disable-cloog-version-check --disable-isl-version-check --enable-cloog-backend=isl --with-host-libstdcxx='-static-libgcc -lstdc++ -lm' --disable-libssp --enable-threads --disable-nls --disable-libmudflap --disable-libgomp --disable-libstdc__-v3 --disable-sjlj-exceptions --disable-shared --disable-tls --disable-libitm --with-float=soft --with-fpu=vfp --with-arch=armv5te --enable-target-optspace --enable-initfini-array --disable-nls --prefix=/tmp/ndk-User/build/toolchain/prefix --with-sysroot=/tmp/ndk-User/build/toolchain/prefix/sysroot --with-binutils-version=2.23 --with-mpfr-version=3.1.1 --with-mpc-version=1.0.1 --with-gmp-version=5.0.5 --with-gcc-version=4.8 --with-gdb-version=7.3.x --with-python=/Users/User/mydroid/ndk/prebuilt/darwin-x86_64/bin/python-config.sh --with-gxx-include-dir=/tmp/ndk-User/build/toolchain/prefix/include/c++/4.8 --with-bugurl=http://source.android.com/source/report-bugs.html --disable-bootstrap --disable-libquadmath --enable-plugins --enable-libgomp --disable-libsanitizer --enable-gold --enable-graphite=yes --with-cloog-version=0.18.0 --with-isl-version=0.11.1 --enable-eh-frame-hdr-for-static --with-arch=armv5te --program-transform-name='s&^&arm-linux-androideabi-&' --enable-gold=default
Thread model: posix
gcc version 4.8 (GCC) 
configure:3565: $? = 0
configure:3554: arm-linux-androideabi-gcc -DANDROID -mandroid -fomit-frame-pointer --sysroot /Users/ben/.buildozer/android/platform/android-ndk-r9c/platforms/android-19/arch-arm -V >&5
arm-linux-androideabi-gcc: error: unrecognized command line option '-V'
arm-linux-androideabi-gcc: fatal error: no input files
compilation terminated.
configure:3565: $? = 1
configure:3554: arm-linux-androideabi-gcc -DANDROID -mandroid -fomit-frame-pointer --sysroot /Users/ben/.buildozer/android/platform/android-ndk-r9c/platforms/android-19/arch-arm -qversion >&5
arm-linux-androideabi-gcc: error: unrecognized command line option '-qversion'
arm-linux-androideabi-gcc: fatal error: no input files
compilation terminated.
configure:3565: $? = 1
configure:3585: checking whether the C compiler works
configure:3607: arm-linux-androideabi-gcc -DANDROID -mandroid -fomit-frame-pointer --sysroot /Users/ben/.buildozer/android/platform/android-ndk-r9c/platforms/android-19/arch-arm -DANDROID -mandroid -fomit-frame-pointer --sysroot /Users/ben/.buildozer/android/platform/android-ndk-r9c/platforms/android-19/arch-arm -DNO_MALLINFO  -lm -L/Users/ben/Library/Application Support/python-for-android/build/libs_collections/unnamed_dist_7/armeabi conftest.c  >&5
arm-linux-androideabi-gcc: error: Support/python-for-android/build/libs_collections/unnamed_dist_7/armeabi: No such file or directory
configure:3611: $? = 1
configure:3649: result: no
configure: failed program was:
| /* confdefs.h */
| #define _GNU_SOURCE 1
| #define _NETBSD_SOURCE 1
| #define __BSD_VISIBLE 1
| #define _BSD_TYPES 1
| #define _DARWIN_C_SOURCE 1
| /* end confdefs.h.  */
| 
| int
| main ()
| {
| 
|   ;
|   return 0;
| }
configure:3654: error: in `/Users/ben/Library/Application Support/python-for-android/build/other_builds/python2/armeabi/python2':
configure:3656: error: C compiler cannot create executables
See `config.log' for more details

## ---------------- ##
## Cache variables. ##
## ---------------- ##

ac_cv_env_CC_set=set
ac_cv_env_CC_value='arm-linux-androideabi-gcc -DANDROID -mandroid -fomit-frame-pointer --sysroot /Users/ben/.buildozer/android/platform/android-ndk-r9c/platforms/android-19/arch-arm'
ac_cv_env_CFLAGS_set=set
ac_cv_env_CFLAGS_value='-DANDROID -mandroid -fomit-frame-pointer --sysroot /Users/ben/.buildozer/android/platform/android-ndk-r9c/platforms/android-19/arch-arm -DNO_MALLINFO'
ac_cv_env_CPPFLAGS_set=
ac_cv_env_CPPFLAGS_value=
ac_cv_env_CPP_set=
ac_cv_env_CPP_value=
ac_cv_env_LDFLAGS_set=set
ac_cv_env_LDFLAGS_value='-lm -L/Users/ben/Library/Application Support/python-for-android/build/libs_collections/unnamed_dist_7/armeabi'
ac_cv_env_LIBS_set=
ac_cv_env_LIBS_value=
ac_cv_env_build_alias_set=set
ac_cv_env_build_alias_value=x86_64-apple-darwin15.3.0
ac_cv_env_host_alias_set=set
ac_cv_env_host_alias_value=arm-eabi
ac_cv_env_target_alias_set=
ac_cv_env_target_alias_value=
ac_cv_prog_CC='arm-linux-androideabi-gcc -DANDROID -mandroid -fomit-frame-pointer --sysroot /Users/ben/.buildozer/android/platform/android-ndk-r9c/platforms/android-19/arch-arm'

## ----------------- ##
## Output variables. ##
## ----------------- ##

AR='arm-linux-androideabi-ar'
ARCH_RUN_32BIT=''
ARFLAGS=''
BASECFLAGS=''
BLDLIBRARY=''
BLDSHARED=''
BUILDEXEEXT=''
CC='arm-linux-androideabi-gcc -DANDROID -mandroid -fomit-frame-pointer --sysroot /Users/ben/.buildozer/android/platform/android-ndk-r9c/platforms/android-19/arch-arm'
CCSHARED=''
CFLAGS='-DANDROID -mandroid -fomit-frame-pointer --sysroot /Users/ben/.buildozer/android/platform/android-ndk-r9c/platforms/android-19/arch-arm -DNO_MALLINFO'
CFLAGSFORSHARED=''
CONFIGURE_MACOSX_DEPLOYMENT_TARGET=''
CONFIG_ARGS=' '\''--host=arm-eabi'\'' '\''--build=x86_64-apple-darwin15.3.0'\'' '\''--prefix=/Users/ben/Library/Application Support/python-for-android/build/other_builds/python2/armeabi/python2/python-install'\'' '\''--enable-shared'\'' '\''--disable-toolbox-glue'\'' '\''--disable-framework'\'' '\''build_alias=x86_64-apple-darwin15.3.0'\'' '\''host_alias=arm-eabi'\'' '\''CC=arm-linux-androideabi-gcc -DANDROID -mandroid -fomit-frame-pointer --sysroot /Users/ben/.buildozer/android/platform/android-ndk-r9c/platforms/android-19/arch-arm'\'' '\''CFLAGS=-DANDROID -mandroid -fomit-frame-pointer --sysroot /Users/ben/.buildozer/android/platform/android-ndk-r9c/platforms/android-19/arch-arm -DNO_MALLINFO'\'' '\''LDFLAGS=-lm -L/Users/ben/Library/Application Support/python-for-android/build/libs_collections/unnamed_dist_7/armeabi'\'''
CPP=''
CPPFLAGS=''
CXX='arm-linux-androideabi-g++ -DANDROID -mandroid -fomit-frame-pointer --sysroot /Users/ben/.buildozer/android/platform/android-ndk-r9c/platforms/android-19/arch-arm'
DEFS=''
DLINCLDIR=''
DLLLIBRARY=''
DYNLOADFILE=''
ECHO_C='\c'
ECHO_N=''
ECHO_T=''
EGREP=''
EXEEXT=''
EXPORT_MACOSX_DEPLOYMENT_TARGET='#'
EXTRAMACHDEPPATH='$(PLATMACPATH)'
EXTRAPLATDIR='$(PLATMACDIRS)'
FRAMEWORKALTINSTALLFIRST=''
FRAMEWORKALTINSTALLLAST=''
FRAMEWORKINSTALLAPPSPREFIX=''
FRAMEWORKINSTALLFIRST=''
FRAMEWORKINSTALLLAST=''
FRAMEWORKUNIXTOOLSPREFIX='/Users/ben/Library/Application Support/python-for-android/build/other_builds/python2/armeabi/python2/python-install'
GNULD=''
GREP=''
HAS_HG=''
HAVE_GETHOSTBYNAME=''
HAVE_GETHOSTBYNAME_R=''
HAVE_GETHOSTBYNAME_R_3_ARG=''
HAVE_GETHOSTBYNAME_R_5_ARG=''
HAVE_GETHOSTBYNAME_R_6_ARG=''
HGBRANCH=''
HGTAG=''
HGVERSION=''
INSTALL_DATA=''
INSTALL_PROGRAM=''
INSTALL_SCRIPT=''
INSTSONAME=''
LDCXXSHARED=''
LDFLAGS='-lm -L/Users/ben/Library/Application Support/python-for-android/build/libs_collections/unnamed_dist_7/armeabi'
LDLAST=''
LDLIBRARY=''
LDLIBRARYDIR=''
LDSHARED=''
LIBC=''
LIBFFI_INCLUDEDIR=''
LIBM=''
LIBOBJS=''
LIBRARY=''
LIBS=''
LIBTOOL_CRUFT=''
LINKCC=''
LINKFORSHARED=''
LIPO_32BIT_FLAGS=''
LN=''
LTLIBOBJS=''
MACHDEP='darwin'
MACHDEP_OBJS=''
MAINCC=''
OBJEXT=''
OPT=''
OTHER_LIBTOOL_OPT=''
PACKAGE_BUGREPORT='http://bugs.python.org/'
PACKAGE_NAME='python'
PACKAGE_STRING='python 2.7'
PACKAGE_TARNAME='python'
PACKAGE_URL=''
PACKAGE_VERSION='2.7'
PATH_SEPARATOR=':'
PKG_CONFIG=''
PYTHONFRAMEWORK=''
PYTHONFRAMEWORKDIR='no-framework'
PYTHONFRAMEWORKIDENTIFIER='org.python.python'
PYTHONFRAMEWORKINSTALLDIR=''
PYTHONFRAMEWORKPREFIX=''
RANLIB='arm-linux-androideabi-ranlib'
RUNSHARED=''
SGI_ABI=''
SHELL='/bin/sh'
SHLIBS=''
SIGNAL_OBJS=''
SO=''
SOVERSION='1.0'
SRCDIRS=''
SVNVERSION=''
THREADHEADERS=''
THREADOBJ=''
TRUE=''
UNICODE_OBJS=''
UNIVERSALSDK=''
UNIVERSAL_ARCH_FLAGS=''
USE_SIGNAL_MODULE=''
USE_THREAD_MODULE=''
VERSION='2.7'
ac_ct_CC=''
bindir='${exec_prefix}/bin'
build_alias='x86_64-apple-darwin15.3.0'
datadir='${datarootdir}'
datarootdir='${prefix}/share'
docdir='${datarootdir}/doc/${PACKAGE_TARNAME}'
dvidir='${docdir}'
exec_prefix='NONE'
host_alias='arm-eabi'
htmldir='${docdir}'
includedir='${prefix}/include'
infodir='${datarootdir}/info'
libdir='${exec_prefix}/lib'
libexecdir='${exec_prefix}/libexec'
localedir='${datarootdir}/locale'
localstatedir='${prefix}/var'
mandir='${datarootdir}/man'
oldincludedir='/usr/include'
pdfdir='${docdir}'
prefix='/Users/ben/Library/Application Support/python-for-android/build/other_builds/python2/armeabi/python2/python-install'
program_transform_name='s,x,x,'
psdir='${docdir}'
sbindir='${exec_prefix}/sbin'
sharedstatedir='${prefix}/com'
sysconfdir='${prefix}/etc'
target_alias=''

## ----------- ##
## confdefs.h. ##
## ----------- ##

/* confdefs.h */
#define _GNU_SOURCE 1
#define _NETBSD_SOURCE 1
#define __BSD_VISIBLE 1
#define _BSD_TYPES 1
#define _DARWIN_C_SOURCE 1

configure: exit 77

````


<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/30498864-building-python2-for-armeabi-fails-due-to-space-character-in-storage_dir-os-x?utm_campaign=plugin&utm_content=tracker%2F11371&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F11371&utm_medium=issues&utm_source=github).
</bountysource-plugin>

--- a/pythonforandroid/archs.py
+++ b/pythonforandroid/archs.py
@@ -61,6 +61,7 @@ class Arch(object):
             ccache = self.ctx.ccache + ' '
             env['USE_CCACHE'] = '1'
             env['NDK_CCACHE'] = self.ctx.ccache
+            env.update({k: v for k, v in environ.items() if k.startswith('CCACHE_')})
 
         print('path is', environ['PATH'])
         cc = find_executable('{command_prefix}-gcc'.format(

--- a/pythonforandroid/build.py
+++ b/pythonforandroid/build.py
@@ -97,6 +97,7 @@ class Context(object):
         self.build_dir = join(self.storage_dir, 'build')
         self.dist_dir = join(self.storage_dir, 'dists')
 
+    def ensure_dirs(self):
         ensure_dir(self.storage_dir)
         ensure_dir(self.build_dir)
         ensure_dir(self.dist_dir)
@@ -155,7 +156,7 @@ class Context(object):
     def ndk_dir(self, value):
         self._ndk_dir = value
 
-    def prepare_build_environment(self, storage_dir, user_sdk_dir, user_ndk_dir,
+    def prepare_build_environment(self, user_sdk_dir, user_ndk_dir,
                                   user_android_api, user_ndk_ver):
         '''Checks that build dependencies exist and sets internal variables
         for the Android SDK etc.
@@ -164,7 +165,7 @@ class Context(object):
 
         '''
 
-        self.setup_dirs(storage_dir)
+        self.ensure_dirs()
 
         if self._build_env_prepared:
             return

--- a/pythonforandroid/build.py
+++ b/pythonforandroid/build.py
@@ -8,7 +8,6 @@ import glob
 import sys
 import re
 import sh
-from appdirs import user_data_dir
 
 from pythonforandroid.util import (ensure_dir, current_directory)
 from pythonforandroid.logger import (info, warning, error, info_notify,
@@ -88,19 +87,21 @@ class Context(object):
         dir = join(self.python_installs_dir, self.bootstrap.distribution.name)
         return dir
 
-    def setup_dirs(self):
+    def setup_dirs(self, storage_dir):
         '''Calculates all the storage and build dirs, and makes sure
         the directories exist where necessary.'''
-        self.root_dir = realpath(dirname(__file__))
-
-        # AND: TODO: Allow the user to set the build_dir
-        self.storage_dir = user_data_dir('python-for-android')
+        self.storage_dir = expanduser(storage_dir)
+        if ' ' in self.storage_dir:
+            raise ValueError('storage dir path cannot contain spaces, please '
+                             'specify a path with --storage-dir')
         self.build_dir = join(self.storage_dir, 'build')
         self.dist_dir = join(self.storage_dir, 'dists')
 
         ensure_dir(self.storage_dir)
         ensure_dir(self.build_dir)
         ensure_dir(self.dist_dir)
+        ensure_dir(join(self.build_dir, 'bootstrap_builds'))
+        ensure_dir(join(self.build_dir, 'other_builds'))
 
     @property
     def android_api(self):
@@ -154,7 +155,7 @@ class Context(object):
     def ndk_dir(self, value):
         self._ndk_dir = value
 
-    def prepare_build_environment(self, user_sdk_dir, user_ndk_dir,
+    def prepare_build_environment(self, storage_dir, user_sdk_dir, user_ndk_dir,
                                   user_android_api, user_ndk_ver):
         '''Checks that build dependencies exist and sets internal variables
         for the Android SDK etc.
@@ -162,6 +163,8 @@ class Context(object):
         ..warning:: This *must* be called before trying any build stuff
 
         '''
+
+        self.setup_dirs(storage_dir)
 
         if self._build_env_prepared:
             return
@@ -434,9 +437,6 @@ class Context(object):
         self.local_recipes = None
         self.copy_libs = False
 
-        # root of the toolchain
-        self.setup_dirs()
-
         # this list should contain all Archs, it is pruned later
         self.archs = (
             ArchARM(self),
@@ -445,9 +445,7 @@ class Context(object):
             ArchAarch_64(self),
             )
 
-        ensure_dir(join(self.build_dir, 'bootstrap_builds'))
-        ensure_dir(join(self.build_dir, 'other_builds'))
-        # other_builds: where everything else is built
+        self.root_dir = realpath(dirname(__file__))
 
         # remove the most obvious flags that can break the compilation
         self.env.pop("LDFLAGS", None)

--- a/pythonforandroid/recipe.py
+++ b/pythonforandroid/recipe.py
@@ -565,9 +565,13 @@ class Recipe(with_metaclass(RecipeMeta)):
                 info('Deleting {}'.format(directory))
                 shutil.rmtree(directory)
 
-    def install_libs(self, arch, lib, *libs):
+    def install_libs(self, arch, *libs):
         libs_dir = self.ctx.get_libs_dir(arch.arch)
-        shprint(sh.cp, '-t', libs_dir, lib, *libs)
+        if not libs:
+            warning('install_libs called with no libraries to install!')
+            return
+        args = libs + (libs_dir,)
+        shprint(sh.cp, *args)
 
     def has_libs(self, arch, *libs):
         return all(map(lambda l: self.ctx.has_lib(arch.arch, l), libs))
@@ -577,8 +581,9 @@ class Recipe(with_metaclass(RecipeMeta)):
         recipe_dirs = []
         if ctx.local_recipes is not None:
             recipe_dirs.append(ctx.local_recipes)
-        recipe_dirs.extend([join(ctx.storage_dir, 'recipes'),
-                            join(ctx.root_dir, "recipes")])
+        if ctx.storage_dir:
+            recipe_dirs.append(join(ctx.storage_dir, 'recipes'))
+        recipe_dirs.append(join(ctx.root_dir, "recipes"))
         return recipe_dirs
 
     @classmethod

--- a/pythonforandroid/toolchain.py
+++ b/pythonforandroid/toolchain.py
@@ -171,8 +171,6 @@ def split_argument_list(l):
 class ToolchainCL(object):
 
     def __init__(self):
-        self._ctx = None
-
         parser = argparse.ArgumentParser(
                 description="Tool for managing the Android / Python toolchain",
                 usage="""toolchain <command> [<args>]
@@ -296,6 +294,8 @@ build_dist
 
         if args.debug:
             logger.setLevel(logging.DEBUG)
+
+        self.ctx = Context()
         self.storage_dir = args.storage_dir
         self.ctx.setup_dirs(self.storage_dir)
         self.sdk_dir = args.sdk_dir
@@ -348,12 +348,6 @@ build_dist
         for line in lines:
             for arg in line:
                 sys.argv.append(arg)
-
-    @property
-    def ctx(self):
-        if self._ctx is None:
-            self._ctx = Context()
-        return self._ctx
 
     def recipes(self, args):
         parser = argparse.ArgumentParser(

--- a/pythonforandroid/toolchain.py
+++ b/pythonforandroid/toolchain.py
@@ -93,8 +93,7 @@ def require_prebuilt_dist(func):
     def wrapper_func(self, args):
         ctx = self.ctx
         ctx.set_archs(self._archs)
-        ctx.prepare_build_environment(storage_dir=self.storage_dir,
-                                      user_sdk_dir=self.sdk_dir,
+        ctx.prepare_build_environment(user_sdk_dir=self.sdk_dir,
                                       user_ndk_dir=self.ndk_dir,
                                       user_android_api=self.android_api,
                                       user_ndk_ver=self.ndk_version)
@@ -298,6 +297,7 @@ build_dist
         if args.debug:
             logger.setLevel(logging.DEBUG)
         self.storage_dir = args.storage_dir
+        self.ctx.setup_dirs(self.storage_dir)
         self.sdk_dir = args.sdk_dir
         self.ndk_dir = args.ndk_dir
         self.android_api = args.android_api
@@ -423,7 +423,7 @@ build_dist
         parser = argparse.ArgumentParser(
                 description="Delete any distributions that have been built.")
         args = parser.parse_args(args)
-        ctx = Context()
+        ctx = self.ctx
         if exists(ctx.dist_dir):
             shutil.rmtree(ctx.dist_dir)
 
@@ -438,7 +438,7 @@ build_dist
         parser = argparse.ArgumentParser(
                 description="Delete all build files (but not download caches)")
         args = parser.parse_args(args)
-        ctx = Context()
+        ctx = self.ctx
         # if exists(ctx.dist_dir):
         #     shutil.rmtree(ctx.dist_dir)
         if exists(ctx.build_dir):
@@ -476,7 +476,7 @@ build_dist
         parser = argparse.ArgumentParser(
                 description="Delete all download caches")
         args = parser.parse_args(args)
-        ctx = Context()
+        ctx = self.ctx
         if exists(ctx.packages_path):
             shutil.rmtree(ctx.packages_path)
 
@@ -641,7 +641,7 @@ build_dist
         python-for-android will internally use for package building, along
         with information about where the Android SDK and NDK will be called
         from.'''
-        ctx = Context()
+        ctx = self.ctx
         for attribute in ('root_dir', 'build_dir', 'dist_dir', 'libs_dir',
                           'ccache', 'cython', 'sdk_dir', 'ndk_dir',
                           'ndk_platform', 'ndk_ver', 'android_api'):
@@ -661,7 +661,7 @@ build_dist
     def distributions(self, args):
         '''Lists all distributions currently available (i.e. that have already
         been built).'''
-        ctx = Context()
+        ctx = self.ctx
         dists = Distribution.get_distributions(ctx)
 
         if dists:

--- a/pythonforandroid/toolchain.py
+++ b/pythonforandroid/toolchain.py
@@ -22,7 +22,7 @@ from functools import wraps
 
 import argparse
 import sh
-
+from appdirs import user_data_dir
 
 from pythonforandroid.recipe import (Recipe, PythonRecipe, CythonRecipe,
                                      CompiledComponentsPythonRecipe,
@@ -93,7 +93,8 @@ def require_prebuilt_dist(func):
     def wrapper_func(self, args):
         ctx = self.ctx
         ctx.set_archs(self._archs)
-        ctx.prepare_build_environment(user_sdk_dir=self.sdk_dir,
+        ctx.prepare_build_environment(storage_dir=self.storage_dir,
+                                      user_sdk_dir=self.sdk_dir,
                                       user_ndk_dir=self.ndk_dir,
                                       user_android_api=self.android_api,
                                       user_ndk_ver=self.ndk_version)
@@ -206,18 +207,23 @@ build_dist
             '--debug', dest='debug', action='store_true',
             help='Display debug output and all build info')
         parser.add_argument(
-            '--sdk_dir', dest='sdk_dir', default='',
+            '--sdk-dir', '--sdk_dir', dest='sdk_dir', default='',
             help='The filepath where the Android SDK is installed')
         parser.add_argument(
-            '--ndk_dir', dest='ndk_dir', default='',
+            '--ndk-dir', '--ndk_dir', dest='ndk_dir', default='',
             help='The filepath where the Android NDK is installed')
         parser.add_argument(
-            '--android_api', dest='android_api', default=0, type=int,
+            '--android-api', '--android_api', dest='android_api', default=0, type=int,
             help='The Android API level to build against.')
         parser.add_argument(
-            '--ndk_version', dest='ndk_version', default='',
+            '--ndk-version', '--ndk_version', dest='ndk_version', default='',
             help=('The version of the Android NDK. This is optional, '
                   'we try to work it out automatically from the ndk_dir.'))
+        parser.add_argument(
+            '--storage-dir', dest='storage_dir',
+            default=self.default_storage_dir,
+            help=('Primary storage directory for downloads and builds '
+                  '(default: {})'.format(self.default_storage_dir)))
 
         # AND: This option doesn't really fit in the other categories, the
         # arg structure needs a rethink
@@ -228,7 +234,7 @@ build_dist
 
         # Options for specifying the Distribution
         parser.add_argument(
-            '--dist_name',
+            '--dist-name', '--dist_name',
             help='The name of the distribution to use or create',
             default='')
         parser.add_argument(
@@ -291,6 +297,7 @@ build_dist
 
         if args.debug:
             logger.setLevel(logging.DEBUG)
+        self.storage_dir = args.storage_dir
         self.sdk_dir = args.sdk_dir
         self.ndk_dir = args.ndk_dir
         self.android_api = args.android_api
@@ -321,6 +328,13 @@ build_dist
         self.ctx.copy_libs = args.copy_libs
 
         getattr(self, args.command)(unknown)
+
+    @property
+    def default_storage_dir(self):
+        udd = user_data_dir('python-for-android')
+        if ' ' in udd:
+            udd = '~/.python-for-android'
+        return udd
 
     def _read_configuration(self):
         # search for a .p4a configuration file in the current directory

